Rakefile

```diff
@@ -22,6 +23,22 @@ task :test do
   Rake::Task['test:core'].invoke
 end
 
+begin
+  require 'minitest/reporters'
```

Comment:
Can you extract this change out to it's own dedicated PR? I think that would help identify some of the issues and focus the PRs.

Reply:
#122 it has dependencies with other changes from this PR though it is failing.